km/km_filesys.c

```diff
@@ -2441,18 +2485,51 @@ km_fs_recover_fdpair(int guestfd[2], int hostfd[2], km_file_how_t how[2], int fl
    // Ensure we don't overwrite hostfd[1] in error.
    if (guestfd[0] >= machine.filesys->nfdmap) {
       km_warn("recover fd %d is invalid", guestfd[0]);
-      return -1;
+      return EINVAL;
    }
    if (guestfd[1] >= machine.filesys->nfdmap) {
       km_warn("recover fd %d is invalid", guestfd[1]);
-      return -1;
+      return EINVAL;
    }
+   int rc;
    if (guestfd[0] == hostfd[1]) {
-      km_fs_recover_fd(guestfd[1], hostfd[1], flags[1], km_get_nonfile_name(hostfd[1]), guestfd[0], how[1]);
-      km_fs_recover_fd(guestfd[0], hostfd[0], flags[0], km_get_nonfile_name(hostfd[0]), guestfd[1], how[0]);
+      rc = km_fs_recover_fd(guestfd[1],
+                            hostfd[1],
+                            flags[1],
+                            km_get_nonfile_name(hostfd[1]),
+                            guestfd[0],
+                            how[1]);
+      if (rc != 0) {
```

Comment:
So recover is changed the same way? Maybe not really needed, cannot really restart recover and hope to succeed.

Reply:
I got carried away.
I need to revert these recover changes.